### PR TITLE
Added project info to report

### PR DESCRIPTION
**Pull request by WackerO**
...and added newline between report author and workflow version. Example report with the new header is attached

[SRP254919_contact.html.zip](https://github.com/nf-core/differentialabundance/files/13349589/SRP254919_contact.html.zip)

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/differentialabundance/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/differentialabundance _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: assets/differentialabundance_report.Rmd [hunk 1]
```diff
@@ -22,6 +22,22 @@ params:
   study_abundance_type: NULL
   report_file: NULL,
   report_title: NULL,
+  report_pi_name: NULL
```
Comment (pinin4fjords) [inline assets/differentialabundance_report.Rmd:25]: Could we just have comma-separated lists please?

File: assets/differentialabundance_report.Rmd [hunk 2]
```diff
@@ -178,13 +194,44 @@ make_params_table <- function(name, pattern = NULL, remove_pattern = FALSE){
...
 subtitle: `r report_subtitle`
 ---
+\
+
+```{r, results='asis', echo=F, eval=T}
+if (!is.null(params$report_pi_name)) {
+  cat(paste0("**Project Members**:  \n"))
+  cat(paste0((params$report_pi_name)))
+}
+# Add further PI info only if name was provided
```
Comment (pinin4fjords) [inline assets/differentialabundance_report.Rmd:211]: Let's just have strings that we split containing contributors, roles etc. I don't think we need to hard-code the concept of a 'PI'.

**Comment by pinin4fjords** [comment]
I also think we should add something to the CSS to place 'contributor blocks' inline, so that we better use the horizontal space.

**Comment by WackerO** [comment]
> I also think we should add something to the CSS to place 'contributor blocks' inline, so that we better use the horizontal space.

You mean so that the different blocks are next to each other? Let me see if I can figure something out

**Comment by WackerO** [comment]
Alright, some changes later: The param `--report_contacts_file` expects a single TXT file with all the PIs and authors and project managers and whatever. As it is, the TXT file in this zip will produce the report in the same zip:
[all_bold.zip](https://github.com/nf-core/differentialabundance/files/13951616/all_bold.zip)

Alternatively, I removed the asterisks from the RMD file and added them directly into the contacts file (see formatted zip). This allows users to add markdown symbols into their contacts to decide themselves how they want that section of the report to be formatted.
I personally find this to be really cool (and as bold/italics can both be done with * characters, it is also fairly simple to use), but of course that would be the "advanced" version and I'm not sure if we should keep this as simple as possible and just print all contacts in bold.
Thoughts?
[formatted.zip](https://github.com/nf-core/differentialabundance/files/13951654/formatted.zip)

**Comment by pinin4fjords** [comment]
Sorry, but I think this has just got over-complicated- now you have a new file format for the contributors to document and maintain for users, plus the parsing logic. 

I suggest something much simpler. A set of parms with comma-separated values:

- report_contributor_names
- report_contributor_role (PI, postdoc, bioinformatics engineer)
- report_contributor_affiliation

Again, don't assume PI, it only makes sense in academia

**Comment by WackerO** [comment]
> Sorry, but I think this has just got over-complicated- now you have a new file format for the contributors to document and maintain for users, plus the parsing logic.
> 
> I suggest something much simpler. A set of parms with comma-separated values:
> 
>     * report_contributor_names
> 
>     * report_contributor_role (PI, postdoc, bioinformatics engineer)
> 
>     * report_contributor_affiliation
> 
> 
> Again, don't assume PI, it only makes sense in academia

Done! As discussed, it is now only a single string param; contribs are separated by `;`, linebreaks are indicated by `\n`

**Comment by pinin4fjords** [comment]
Also, an example report with this please?

**Comment by WackerO** [comment]
> Also, an example report with this please?

Sure, here you go!
[SRP254919.html.zip](https://github.com/nf-core/differentialabundance/files/14034289/SRP254919.html.zip)

File: assets/differentialabundance_report.Rmd [hunk 3]
```diff
@@ -250,6 +252,33 @@ cat(paste0("
...
+      cat(paste0("<div class='div-row'>"))                                # For 1., 3. etc. entry open surrounding div that will contain 2 contributors side-by-side
+    }
+    cat(paste0("<div class='div-column'>", contributors[r], "</div>"))    # Create inner div containing the PI info itself
+    if (r %% 2 == 0 || r == length(contributors)) {
+      cat(paste0("</div>"))                                               # For 2., 4. etc. entry and for last entry, close outer div
+    }
+  }
+} else if (length(contributors) == 1) {
+  cat(paste0("", contributors, ""))
+}
```
Comment (pinin4fjords) [inline assets/differentialabundance_report.Rmd:279]:
```suggestion
contributors <- gsub("\n", "<br>", params$report_contributors, fixed = TRUE)
contributors <- lapply(simpleSplit(contributors, ";"), function(s) {
  splt <- simpleSplit(s, "<br>")
  paste0("**", head(splt, 1), "**<br>", paste(tail(splt, -1), collapse = "<br>"))
})

for (r in seq_along(contributors)) {
  if (r %% 2 == 1) cat("<div class='div-row'>")
  cat(paste0("<div class='div-column'>", contributors[r], "</div>"))
  if (r %% 2 == 0 || r == length(contributors)) cat("</div>")
}
```

Suggested simplification from ChatGPT. Tested, and it works. The difference is that we get HTML wrapping even with a single contributor, but I think that's better for consistency.

**Review by pinin4fjords — APPROVED** [review]
Looks good now- much more concise. Thanks!

**Pull request merged**